### PR TITLE
Files parallel

### LINKED ISSUE #313: "Files" aren't pickled well in internalParallel runs
--------
Issue Description
--------
##### What did you expect to see happen?
No special treatment for GenericCodeInterface codes using Files

##### What did you see instead?
Error:
```python
A fatal error has occured during the function execution
Traceback (most recent call last):
  File "/home/talbpaul/projects/raven/framework/contrib/pp/ppworker.py", line 107, in run
    __args = pickle.loads(__sargs)
  File "/home/talbpaul/anaconda2/envs/raven_libraries/lib/python2.7/pickle.py", line 1388, in loads
    return Unpickler(file).load()
  File "/home/talbpaul/anaconda2/envs/raven_libraries/lib/python2.7/pickle.py", line 864, in load
    dispatch[key](self)
  File "/home/talbpaul/anaconda2/envs/raven_libraries/lib/python2.7/pickle.py", line 1223, in load_build
    setstate(state)
  File "/home/talbpaul/projects/raven/framework/Files.py", line 115, in __setstate__
    self.updateFilename()
AttributeError: 'UserGenerated' object has no attribute 'updateFilename'
```
##### Do you have a suggested fix for the development team?
A test needs to be added to cover this behavior, and fixes made so it works.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users. __cannot cause wrong results__
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed? *Yes*
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.) *Yes*
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?  *Unlikely to impact users, no email required*
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)? *No stable branch*
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided? *Will be closed with merge request*

Close reviewed by @joshua-cogliati-inl

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? 
Closes #313.  

##### What are the significant changes in functionality due to this change request?
 - `Csv_loader` renamed to `CsvLoader` for consistency
 - `CsvLoader` is now correctly added to `Models.Code.mods` list
 - Fixed pickling of "Files" objects to work consistently
 - Also adds a user guide entry on the basic statistics to xml to dataobject by way of the raven output postprocessor.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. *Done*
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).  *No change to input*
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. *Yes*
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. *Tests added*
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. *No*
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. *Done*
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? *No changes*

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
Sure, thanks!

File: doc/user_guide/statisticalAnalysisExample.tex [hunk 1]
```diff
@@ -192,3 +192,4 @@ \section{Statistical Analysis through RAVEN}
 \end{table}
 \end{landscape}
 
+\input{statisticsToDataObject.tex}
```
Comment (joshua-cogliati-inl) [inline doc/user_guide/statisticalAnalysisExample.tex]: Just to double check, is this supposed to be part of the merge?

File: framework/Models/Code.py [hunk 2]
```diff
@@ -338,7 +338,7 @@ def createNewInput(self,currentInput,samplerType,**kwargs):
 
     return (newInput,kwargs)
 
-  def __expandForWindows(self, origCommand):
+  def _expandForWindows(self, origCommand):
```
Comment (joshua-cogliati-inl) [inline framework/Models/Code.py]: This doesn't seem to be used outside the class, so is there a reason it was switched from private to protected?

File: doc/user_guide/statisticalAnalysisExample.tex (hunk 1 above)
Comment (PaulTalbot-INL) [reply doc/user_guide/statisticalAnalysisExample.tex]: It was not originally intended to, but there's not a better merge for it to go in with, and the example should be added.  I think the TeX file is there, too.

File: tests/framework/user_guide/StatisticalAnalysis/comparingSamplers.xml [hunk 3]
```diff
@@ -0,0 +1,233 @@
+<?xml version="1.0" ?>
```
Comment (joshua-cogliati-inl) [inline tests/framework/user_guide/StatisticalAnalysis/comparingSamplers.xml]: This file doesn't seem to be tested.

File: doc/user_guide/statisticalAnalysisExample.tex (hunk 1 above)
Comment (joshua-cogliati-inl) [reply doc/user_guide/statisticalAnalysisExample.tex]: Okay, thanks.

File: doc/user_guide/statisticsToDataObject.tex [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+\end{figure}
+\begin{figure}[h!]
+  \centering
+  \includegraphics[width=0.7\linewidth]{../../tests/framework/user_guide/StatisticalAnalysis/comparingSamplers/varPlotter_scatter}
+  \caption{Sampler Comparison Plot, Variance}
+\end{figure}
+
+
+%%%%%%%%%%%%%%%%%%% OLD
+%In order to accomplish these tasks, the following RAVEN \textbf{Entities} need to be defined (these are
```
Comment (joshua-cogliati-inl) [inline doc/user_guide/statisticsToDataObject.tex]: Should this section be kept?

File: doc/user_manual/model.tex [hunk 5]
```diff
@@ -1237,18 +1237,6 @@ \subsubsection{Method: \texttt{def run}}
 \input{postprocessor.tex}
 
 
-\paragraph{Method: dataObjectLabelFilter}
-The \xmlNode{HSPS} Post-Processor performs a filtering of the dataObject. This particular filtering is based on the labels generated by any clustering algorithm.
```
Comment (joshua-cogliati-inl) [inline doc/user_manual/model.tex]: Is this information available somewhere else or is it outdated? (As in why is it removed?)

File: framework/Models/Code.py (hunk 2 above)
Comment (PaulTalbot-INL) [reply framework/Models/Code.py]: As a private member it wasn't recognizing it when called later on.  I don't know why that is, but this change fixed it.

File: doc/user_manual/model.tex (hunk 5 above)
Comment (PaulTalbot-INL) [reply doc/user_manual/model.tex]: I don't recall; I will look into it.

File: framework/Models/Code.py (hunk 2 above)
Comment (joshua-cogliati-inl) [reply framework/Models/Code.py]: Okay.

File: doc/user_manual/model.tex (hunk 5 above)
Comment (PaulTalbot-INL) [reply doc/user_manual/model.tex]: Oh, yes, this appeared in two places in the input manual, and on instruction from Diego I removed it from this one.

File: doc/user_guide/statisticsToDataObject.tex (hunk 4 above)
Comment (PaulTalbot-INL) [reply doc/user_guide/statisticsToDataObject.tex]: Removed.

File: tests/framework/user_guide/StatisticalAnalysis/comparingSamplers.xml (hunk 3 above)
Comment (PaulTalbot-INL) [reply tests/framework/user_guide/StatisticalAnalysis/comparingSamplers.xml]: Fixed.

**Comment by PaulTalbot-INL** [comment]
Adding heavy testing to this PR since some of the test files use it.

File: tests/framework/user_guide/StatisticalAnalysis/tests [hunk 6]
```diff
@@ -3,7 +3,13 @@
...
     input = 'statisticalAnalysis.xml'
     UnorderedXml = 'StatisticalAnalysis/output_stat_analysis_mc.xml'
-    required_libraries = 'PIL'
+    heavy = true
+    rel_err = 1e-3
+  [../]
+  [./ComparingSamplers]
+    type = 'RavenFramework'
+    input = 'comparingSamplers.xml'
+    UnorderedXml = 'comparingSamplers/output_stat_analysis_mc.xml comparingSamplers/output_stat_analysis_lhs.xml comparingSamplers/output_stat_analysis_grid.xml'
```
Comment (joshua-cogliati-inl) [inline tests/framework/user_guide/StatisticalAnalysis/tests]: Where are the gold files?

**Comment by joshua-cogliati-inl** [comment]
```FAILED (Gold file does not exist: /home/moosetest/civet/client_root_0/raven/tests/framework/user_guide/StatisticalAnalysis/gold/comparingSamplers/output_stat_analysis_mc.xmlGold file does not exist: /home/moosetest/civet/client_root_0/raven/tests/framework/user_guide/StatisticalAnalysis/gold/comparingSamplers/output_stat_analysis_lhs.xmlGold file does not exist: /home/moosetest/civet/client_root_0/raven/tests/framework/user_guide/StatisticalAnalysis/gold/comparingSamplers/output_stat_analysis_grid.xml) (reprint)
```

**Comment by PaulTalbot-INL** [comment]
The heavy tests are failing because the plots are a different size.  Is there a way to know when the heavy test machine changes its matplotlib settings?

**Comment by joshua-cogliati-inl** [comment]
Maybe we should set the sizes and the dpi so we don't have them change.  See: https://stackoverflow.com/questions/10041627/how-to-make-pylab-savefig-save-image-for-maximized-window-instead-of-default

**Comment by PaulTalbot-INL** [comment]
I thought about that, too.  I don't yet know if there's a combination of keywords we can pass through our interface to convince matplotlib to fix the size.  I'll look into it.

**Comment by PaulTalbot-INL** [comment]
Due to the complications incurred when adding the user guide change to this PR, I'm going to rescind it and open a new request for the Files changes only. See #326.

**Pull request closed without merging**